Update `StorageConfig` to handle multiple auth methods (#1140)
ethyca/fidesops: docs/fidesops/docs/guides/storage.md

Additional params needed for S3:

> - `auth_method`: Authentication method for creating a session with S3. 

Reviewer: It looks like the only accepted methods are "automatic" or "secret_keys"? We should definitely note that here.
```suggestion
- `auth_method`: Authentication method for creating a session with S3, either `automatic` or `secret_keys`.
```

> - `auth_method`: Authentication method for creating a session with S3. 

Author: @sanders41 Are you referring to the 8 methods mentioned on the boto3 documentation? [All boto3 auth methods](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html)

> - `auth_method`: Authentication method for creating a session with S3. 

Reviewer 2: Right, does the 2 ways you have here work with all 8? Or will it only work with certain ones?

> - `auth_method`: Authentication method for creating a session with S3. 

Author: It doesn't work with all 8. I haven't verified every option that `automatic` covers but it should handle anything supports automatically creating a session based on the container environment.

```

Params needed for S3:
- `aws_access_key_id`: AWS access key id, obtained from AWS console.
- `aws_secret_access_key`: AWS secret access key, obtained from AWS console.

> Fidesops also supports automatically creating a session for S3. When this authentication method is selected no secrets need to be provided. Boto3 will look for credentials on the server.

Reviewer: ```suggestion
Fidesops also supports automatically creating a session for S3. If your `auth_method` is set to `automatic`, no secrets need to be provided. Boto3 will look for credentials on the server.
```

> Fidesops also supports automatically creating a session for S3. When this authentication method is selected no secrets need to be provided. Boto3 will look for credentials on the server.

Author: @conceptualshark How does this look for pointing users to the boto3 docs?

> Please refer to the boto3 [credentials](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html) documentation for how to configure credentials for AWS.
